cmd/versionUpgrade.go

```diff
@@ -0,0 +1,58 @@
+package cmd
+
+import (
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+
+	"github.com/spf13/cobra"
+)
+
+var downloadURL = "https://github.com/karimra/gnmic/raw/master/install.sh"
+
+// upgradeCmd represents the version command
+var upgradeCmd = &cobra.Command{
+	Use:   "upgrade",
+	Short: "upgrade gnmic to latest available version",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		f, err := ioutil.TempFile("", "gnmic")
+		defer os.Remove(f.Name())
+		if err != nil {
+			log.Fatalf("Failed to create temp file %s\n", err)
+		}
+		downloadFile(downloadURL, f)
+
+		c := exec.Command("bash", f.Name())
+		c.Stdout = os.Stdout
+		c.Stderr = os.Stderr
+		err = c.Run()
+		if err != nil {
+			log.Fatalf("Upgrade failed: %s\n", err)
+		}
+	},
+}
+
+// downloadFile will download a file from a URL and write its content to a file
+func downloadFile(url string, file *os.File) error {
+	// Get the data
+	resp, err := http.Get(url)
```

Comment:
better use a custom http client here, with a timeout (10s ?) .
The default http client doesn't have one

Reply:
addressed in 7ec3633